src/api.js

```diff
@@ -0,0 +1,182 @@
+let debug = require('debug')('events:api');
+let APIBuilder = require('taskcluster-lib-api');
+let taskcluster = require('taskcluster-client');
+let uuid = require('uuid');
+let _ = require('lodash');
+
+let builder = new APIBuilder({
+  title: 'AMQP Messages API Documentation',
+  description: [
+    'This service is responsible for making pulse messages accessible',
+    'from browsers and cli. There are API endpoints to',
+    'bind / unbind to an exchange and pause / resume listening from a queue',
+  ].join('\n'),
+  projectName: 'taskcluster-evnets',
+  serviceName: 'events',
+  version: 'v1',
+  context: ['connection'],
+});
+
+// Returns JSON.parse(bindings) if everything goes well
+//   {"bindings" : [ 
+//     {"exchange" :  "a/b/c", "routingKey" : "a.b.c"},
+//     {"exchange" :  "x/y/z", "routingKey" : "x.y.z"},
+//   ]};
+var parseAndValidateBindings = function(bindings) {
+  return new Promise((resolve, reject) => {
+    try {
+      let json_bindings = JSON.parse(bindings);
+      if (String(Object.keys(json_bindings)) !== String(['bindings'])) {
+        throw new Error('The json query should have only one key i.e. `bindings`.');
+      }  
+
+      // Reduce json_bindings to an array of exchanges.
+      json_bindings = json_bindings.bindings;
+      if (!Array.isArray(json_bindings)) {
+        throw new Error('Bindings must be an array of {exchange, routingKey}');
+      }
+      json_bindings.map(binding => {
+        let keys = Object.keys(binding);
+        if (keys.length !=2 || !binding.hasOwnProperty('routingKey') || !binding.hasOwnProperty('exchange')) {
+          throw new Error('Each binding must have only two fields - exchange and routingKey');
+        }
```

Comment:
Instead of always checking that no other key properties are there, could we simply ignore fields that don't do anything?

```js
if(!('routingKey' in binding) || !('exchange' in binding)) {
  throw new Error('Binding must include the `exchange` and `routingKey` fields');
}
```

Reply:
Yes I don't see any downside to that.
We can even ignore fields other than `bindings` at the previous level

Reply:
I made this change and skipped the test which now fails.